bot.py

```
@@ -24,6 +25,60 @@
+        **kwargs: Keyword arguments to pass to the function
+    
+    Returns:
+        The result of the function call, or None if all retries failed
+    """
+    last_exception = None
+    
+    for attempt in range(max_retries + 1):
+        try:
+            return await func(*args, **kwargs)
+        except (aiohttp.ClientTimeout, aiohttp.ClientError, asyncio.TimeoutError) as e:
+            last_exception = e
```

Ruff (0.11.9)

Local variable `last_exception` is assigned to but never used

Remove assignment to unused variable `last_exception`

(F841)